### PR TITLE
Moved strided slide ops from reference to its own.

### PR DESCRIPTION
**Pull request by amitsrivastava78**
This is to remove the gemmlowp dependency.

### CODE REVIEW

File: tensorflow/lite/toco/graph_transformations/quantize.cc [hunk 1]
```diff
@@ -62,8 +62,9 @@ bool SupportsQuantization(const Operator& op) {
          type == OperatorType::kLessEqual || type == OperatorType::kSelect ||
          type == OperatorType::kArgMax || type == OperatorType::kRelu ||
          type == OperatorType::kRelu1 || type == OperatorType::kRelu6 ||
-         type == OperatorType::kShape || type == OperatorType::kExpandDims ||
-         type == OperatorType::kPack || type == OperatorType::kTopK_V2 ||
+         type == OperatorType::kLeakyRelu || type == OperatorType::kShape ||
```
Comment (karimnosseir) [inline tensorflow/lite/toco/graph_transformations/quantize.cc]: This is not related to the strided slice, does it ?

File: tensorflow/lite/kernels/internal/reference/strided_stride.h [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```
Comment (karimnosseir) [inline tensorflow/lite/kernels/internal/reference/strided_stride.h]: file should be strided_slice not strided_stride i think ?

**Review by karimnosseir — CHANGES_REQUESTED** [review]
Not clear to me the goal from this change. You're saying removing gemmlowp dependency, can you explain how.

Thanks

File: tensorflow/lite/toco/graph_transformations/quantize.cc (hunk 1 above)
Comment (amitsrivastava78) [reply tensorflow/lite/toco/graph_transformations/quantize.cc]: @karimnosseir , thanks or the review, i have removed this part.

Regards
Amit

File: tensorflow/lite/kernels/internal/reference/strided_stride.h (hunk 2 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/internal/reference/strided_stride.h]: @karimnosseir , thanks for the review, i have renamed the file.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@karimnosseir , I am trying to port strided slice to Micro kernel for our HiSi board, For this i wanted to remove the strided slice dependency first.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@karimnosseir , can you please review the changes, a my other changes are dependent on these.

Regards
Amit

**Review by petewarden — DISMISSED** [review]

**Comment by amitsrivastava78** [comment]
@petewarden thanks for approving the PR, there was one sanity build failure as the names in the BUILD file was not sorted, i have fixed it, kindly approve it again. Sorry for the trouble.

Also i want to port strided slide along with mul and add (for batch norm) on TFlite micro, let me know your views on the same.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@petewarden can you please re approve the PR.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@petewarden , can you pls have alook at the PR and re approve it.

Regards
Amit

**Review by karimnosseir — DISMISSED** [review]

File: tensorflow/lite/kernels/internal/reference/strided_slice.h [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          *out_ptr++ = input_data[Offset(input_shape, in_b, in_h, in_w, in_d)];
+        }
+      }
+    }
+  }
+}
+}  // namespace reference_ops
+}  // namespace tflite
+
+#endif
```
Comment (rthadur) [inline tensorflow/lite/kernels/internal/reference/strided_slice.h]: CheckLint found errors.
#endif line should be "#endif  // TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_STRIDED_SLICE_H_"  [build/header_guard] [5]
	//depot/google3/third_party/tensorflow/lite/kernels/internal/reference/strided_slice.h:80

**Review by rthadur — CHANGES_REQUESTED** [review]

File: tensorflow/lite/kernels/internal/reference/strided_slice.h (hunk 3 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/internal/reference/strided_slice.h]: @rthadur , thanks for the update, i have resolved the issue.


Regards
Amit

**Comment by amitsrivastava78** [comment]
@karimnosseir , sorry to bug you again, there was one check lint issue which is resolved now, can you please re-approve the PR.

Regards
Amit

**Review by karimnosseir — APPROVED** [review]

**Comment by rthadur** [comment]
Changes are committed waiting for auto merge to happen.

**Comment by mihaimaruseac** [comment]
Seems automerge is not happening but the changes are now committed so we can close this. Thank you for the PR.
